ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixClusterManagerTest.java

```diff
@@ -234,6 +232,82 @@ public void after() {
     }
   }
 
+  /**
+   * Test the case where replicas from same partition have different capacities (which should block the startup)
+   * @throws Exception
+   */
+  @Test
+  public void inconsistentReplicaCapacityTest() throws Exception {
+    assumeTrue(listenCrossColo);
+    clusterManager.close();
+    metricRegistry = new MetricRegistry();
+    String staticClusterName = "TestOnly";
+    File tempDir = Files.createTempDirectory("helixClusterManagerTest").toFile();
+    tempDir.deleteOnExit();
+    String tempDirPath = tempDir.getAbsolutePath();
+    String testHardwareLayoutPath = tempDirPath + File.separator + "hardwareLayoutTest.json";
+    String testPartitionLayoutPath = tempDirPath + File.separator + "partitionLayoutTest.json";
+    String testZkLayoutPath = tempDirPath + File.separator + "zkLayoutPath.json";
+
+    // initialize test hardware layout and partition layout, create mock helix cluster for testing.
+    TestHardwareLayout testHardwareLayout1 = constructInitialHardwareLayoutJSON(staticClusterName);
+    TestPartitionLayout testPartitionLayout1 = constructInitialPartitionLayoutJSON(testHardwareLayout1, 3, localDc);
+    JSONObject zkJson = constructZkLayoutJSON(dcsToZkInfo.values());
+    Utils.writeJsonObjectToFile(zkJson, testZkLayoutPath);
+    Utils.writeJsonObjectToFile(testHardwareLayout1.getHardwareLayout().toJSONObject(), testHardwareLayoutPath);
+    Utils.writeJsonObjectToFile(testPartitionLayout1.getPartitionLayout().toJSONObject(), testPartitionLayoutPath);
+    MockHelixCluster testCluster =
+        new MockHelixCluster("AmbryTest-", testHardwareLayoutPath, testPartitionLayoutPath, testZkLayoutPath);
+
+    List<DataNode> initialNodes = testHardwareLayout1.getAllExistingDataNodes();
+    Partition partitionToTest = (Partition) testPartitionLayout1.getPartitionLayout().getPartitions(null).get(0);
+
+    // add a new node into cluster
+    testHardwareLayout1.addNewDataNodes(1);
+    Utils.writeJsonObjectToFile(testHardwareLayout1.getHardwareLayout().toJSONObject(), testHardwareLayoutPath);
+    DataNode newAddedNode =
+        testHardwareLayout1.getAllExistingDataNodes().stream().filter(n -> !initialNodes.contains(n)).findAny().get();
+    System.out.println("new Added Node instance :" + newAddedNode.getHostname() + "_" + newAddedNode.getPort());
```

Comment:
minor: use logger.info() ?

Reply:
Good catch, I should remove this (it was for debugging unit test).